server/service/src/invoice/prescription/delete/mod.rs

```diff
@@ -0,0 +1,155 @@
+use repository::{ActivityLogType, InvoiceRowRepository, RepositoryError};
+
+pub mod validate;
+
+use validate::validate;
+
+use crate::{
+    activity_log::activity_log_entry,
+    invoice::common::get_lines_for_invoice,
+    invoice_line::stock_out_line::{
+        delete_stock_out_line, DeleteStockOutLine, DeleteStockOutLineError, StockOutType,
+    },
+    service_provider::ServiceContext,
+};
+
+type OutError = DeletePrescriptionError;
+
+pub fn delete_prescription(
+    ctx: &ServiceContext,
+    id: String,
+) -> Result<String, DeletePrescriptionError> {
+    let invoice_id = ctx
+        .connection
+        .transaction_sync(|connection| {
+            validate(&id, &ctx.store_id, &connection)?;
+
+            let lines = get_lines_for_invoice(connection, &id)?;
+            for line in lines {
+                delete_stock_out_line(
+                    ctx,
+                    DeleteStockOutLine {
+                        id: line.invoice_line_row.id.clone(),
+                        r#type: Some(StockOutType::Prescription),
+                    },
+                )
+                .map_err(|error| OutError::LineDeleteError {
+                    line_id: line.invoice_line_row.id,
+                    error,
+                })?;
+            }
+
+            activity_log_entry(
+                &ctx,
+                ActivityLogType::InvoiceDeleted,
+                Some(id.to_owned()),
+                None,
+            )?;
+
+            match InvoiceRowRepository::new(&connection).delete(&id) {
+                Ok(_) => Ok(id.clone()),
+                Err(error) => Err(OutError::DatabaseError(error)),
+            }
+        })
+        .map_err(|error| error.to_inner_error())?;
+
+    Ok(invoice_id)
+}
+
+#[derive(Debug, PartialEq, Clone)]
+
+pub enum DeletePrescriptionError {
+    InvoiceDoesNotExist,
+    DatabaseError(RepositoryError),
+    NotThisStoreInvoice,
+    CannotEditFinalised,
+    LineDeleteError {
+        line_id: String,
+        error: DeleteStockOutLineError,
+    },
+    NotAPrescription,
```

Comment:
A query: In the issue this is `NotAPrescriptionInvoice`. I like `NotAPrescription` but it is inconsistent with `NotThisStoreInovice`